CSharpMath.Editor/Extensions/MathList.cs

```diff
@@ -81,11 +81,25 @@ partial class Extensions {
           throw new SubIndexTypeMismatchException(index);
       }
     }
-
+  public static MathListIndex? PreviousOrBeforeWholeList(MathListIndex index) {
```

Comment:
Should this be an extension method?

Reply:
Perhaps. I tried to move it inside the Delete method because the xml comment said "only use from Delete..." but it also needed to be used in DeleteBackwards in MathKeyboard.